### PR TITLE
Only allow host-relative LDConfig paths

### PR DESCRIPTION
**Pull request by elezar**
This change only allows host-relative LDConfig paths (i.e. the `nvidia-container-cli.ldconfig` setting starts with an `@`)

An `allow-ldconfig-from-container` feature flag is added to allow for this
the default behaviour to be changed.

This feature can be toggled by running:
```
sudo nvidia-ctk config --in-place --set features.allow-ldconfig-from-container=true
```
or by including `allow-config-from-container` in the `NVIDIA_CONTAINER_TOOLKIT_OPT_IN_FEATURES` envvar.

### CODE REVIEW

File: internal/config/cli.go [hunk 1]
```diff
@@ -47,7 +47,7 @@ func (c *ContainerCLIConfig) NormalizeLDConfigPath() string {
 // NormalizeLDConfigPath returns the resolved path of the configured LDConfig binary.
 // This is only done for host LDConfigs and is required to handle systems where
 // /sbin/ldconfig is a wrapper around /sbin/ldconfig.real.
-func NormalizeLDConfigPath(path string) string {
+func NormalizeLDConfigPath(path string) (rpath string) {
```
Comment (cdesiniotis) [inline internal/config/cli.go:50]: Why is this change required?

File: internal/config/config_test.go [hunk 2]
```diff
@@ -293,6 +340,7 @@ func TestGetConfig(t *testing.T) {
 
 	for _, tc := range testCases {
 		t.Run(tc.description, func(t *testing.T) {
+			defer setGetLdConfigPathForTest()
```
Comment (cdesiniotis) [inline internal/config/config_test.go:343]: Question: why do we `defer` the execution of this function call? Don't we need to set the ldconfig path for the test case we are about to execute?

Comment (elezar) [reply internal/config/config_test.go:343]: You're right. This should be:
```suggestion
			defer setGetLdConfigPathForTest()()
```

**Review by cdesiniotis — APPROVED** [review]

**Review by klueska — DISMISSED** [review]
1. Let's continue to allow host-relative path to ldconfig even without the override
2. Let's then change the override flag to be called `allow-ldconfig-from-container`

**Comment by elezar** [comment]
> Question -- is it possible to include a comment in the default NVIDIA Container Toolkit config file that gets installed at /etc/nvidia-container-runtime/config.toml which documents the new feature flag we are adding?

The features are currently `omitempty`, meaning that if none of them are set we would not show them. I can look at adding a comment similar to the docstring to the `ldconfig` option though.

**Review by cdesiniotis — APPROVED** [review]
